codex-rs/core/src/config.rs

```diff
@@ -1024,7 +1052,47 @@ impl Config {
 
         let features = Features::from_config(&cfg, &config_profile, feature_overrides);
 
-        let sandbox_policy = cfg.derive_sandbox_policy(sandbox_mode);
+        let resolved_cwd = {
+            use std::env;
+
+            match cwd {
+                None => {
+                    tracing::info!("cwd not set, using current dir");
+                    env::current_dir()?
+                }
+                Some(p) if p.is_absolute() => p,
```

Comment:
I think we may want to introduce the use of https://crates.io/crates/path-absolutize.

Note that, perhaps surprisingly, this is `true`:

```rust
assert!(Path::new("/foo/../bar").is_absolute());
```

absolute is not the same as normalized (or canonicalized)

https://crates.io/crates/path-absolutize also normalizes (but does not canonicalize, which I think is OK because sometimes people use symlinks and don't want them canonicalized, though that can be confusing at times...)